cf/service_visibilities.go

```diff
@@ -138,27 +140,24 @@ func (pc *PlatformClient) getBrokersByName(ctx context.Context, names []string)
 			query := queryBuilder{}
 			query.set("name", brokerNames)
 			brokers, err := pc.client.ListServiceBrokersByQuery(query.build(ctx))
-
 			mutex.Lock()
 			defer mutex.Unlock()
 			if err != nil {
-				if errorOccured == nil {
-					errorOccured = err
-				}
-			} else if errorOccured == nil {
+				errorOccured.Add(err)
+			} else if errorOccured.Len() == 0 {
```

Comment:
What do you think of stopping all the goroutines that fetch visibilities in case of a failure and abort the resync. Now we will wait for all the goroutines to finish fetching their chunks and after that we fail the resync and notifications are stored in this timeframe. Maybe fail fast is better so we can minimize the time of notifications being stored.

Reply:
This should be addressed with separate change because it requires a bigger changes to the synchronisation logic.